pkg/controllers/disruption/types.go

```diff
@@ -140,46 +140,34 @@ func (c Command) Decision() Decision {
 	}
 }
 
-func (c Command) String() string {
-	var buf bytes.Buffer
+func (c Command) LogValues() []any {
 	podCount := lo.Reduce(c.candidates, func(_ int, cd *Candidate, _ int) int { return len(cd.reschedulablePods) }, 0)
-	fmt.Fprintf(&buf, "%s, terminating %d nodes (%d pods) ", c.Decision(), len(c.candidates), podCount)
-	for i, old := range c.candidates {
-		if i != 0 {
-			fmt.Fprint(&buf, ", ")
+
+	candidateNodes := lo.Map(c.candidates, func(candidate *Candidate, _ int) interface{} {
+		return map[string]interface{}{
+			"Node":          klog.KObj(candidate.Node),
+			"NodeClaim":     klog.KObj(candidate.NodeClaim),
+			"instance-type": candidate.Labels()[corev1.LabelInstanceTypeStable],
+			"capacity-type": candidate.Labels()[v1.CapacityTypeLabelKey],
 		}
-		fmt.Fprintf(&buf, "%s", old.Name())
-		fmt.Fprintf(&buf, "/%s", old.Labels()[corev1.LabelInstanceTypeStable])
-		fmt.Fprintf(&buf, "/%s", old.Labels()[v1.CapacityTypeLabelKey])
-	}
-	if len(c.replacements) == 0 {
-		return buf.String()
-	}
-	odNodeClaims := 0
-	spotNodeClaims := 0
-	for _, nodeClaim := range c.replacements {
-		ct := nodeClaim.Requirements.Get(v1.CapacityTypeLabelKey)
-		if ct.Has(v1.CapacityTypeOnDemand) {
-			odNodeClaims++
+	})
+	replacementNodes := lo.Map(c.replacements, func(replacement *scheduling.NodeClaim, _ int) interface{} {
+		ct := replacement.Requirements.Get(v1.CapacityTypeLabelKey)
+		m := map[string]interface{}{
+			"capacity-type": lo.Ternary[string](ct.Has(v1.CapacityTypeSpot), v1.CapacityTypeSpot, v1.CapacityTypeOnDemand),
 		}
-		if ct.Has(v1.CapacityTypeSpot) {
-			spotNodeClaims++
+		if len(c.replacements) == 1 {
+			m["instance-types"] = scheduling.InstanceTypeList(replacement.InstanceTypeOptions)
 		}
+		return m
+	})
+
+	return []any{
+		"decision", c.Decision(),
+		"candidate-count", len(candidateNodes),
```

Comment:
Do we just want to call candidates "nodes"?

Reply:
Updated with `disrupted-node-count`